### ISSUE TITLE
Make it easier to replace the default (transient) lifestyle

### ISSUE THREAD

**Issue by dotnetjunkie** [simpleinjector/SimpleInjector#313]
Replacing the default (transient) lifestyle can be done by implementing a custom `ILifestyleSelectionBehavior`. I would like to simplify this by either creating an out-of-the-box `CustomLifestyleSelectionBehavior` class that allows wrapping an alternative lifestyle, or some practical `Container.Options.SetDefaultLifestyle(Lifestyle)` method.

**Comment by dotnetjunkie** [comment]
Even easier would it be to add a `DefaultLifestyle` property alongside the `DefaultScopedLifestyle` property that already exists in the `ContainerOptions` object. This new `DefaultLifestyle` can be set to `Lifestyle.Transient` by default and the default `ILifestyleSelectionBehavior` implementation can forward to this new property instead of directly returning `Lifestyle.Transient`.

Common usage would be as follows:

``` c#
var container = new Container();
// Everything as singleton by default
container.Options.DefaultLifestyle = Lifestyle.Singleton;
// Async scoped as default scoped
containreer.Options.DefaultScopedLifestyle = new ExecutionContextScopeLifestyle();

// Registers as singleton
container.Register<IUserContext, AspNetUserContext>();

var contextProvider = Lifestyle.Scoped.CreateProducer<MyAppDbContext>(container);
container.RegisterSingleton<IUnitOfWork>(new DbContextUnitOfWork(contextProvider.GetInstance));
```

``` c#
var container = new Container();
// Everything as Scoped by default.
container.Options.DefaultLifestyle = new ExecutionContextScopeLifestyle();

container.Register<IUserContext, AspNetUserContext>();
```

``` c#
var container = new Container();
// Async scoping as scoped lifestyle.
container.Options.DefaultScopedLifestyle = new ExecutionContextScopeLifestyle();
// Custom lifestyle that will dispose transient instances when a scope ends.
container.Options.DefaultLifestyle = 
    new DisposableTransientLifestyle(container.Options.DefaultScopedLifestyle);
```

**Comment by TheBigRic** [comment]
Very clean! 

This makes it far easier to understand than implementing the `ILifestyleSelectionBehaviour` and is in sync with the `DefaultScopedLifestyle` property on the `ContainerOptions`

**Comment by dotnetjunkie** [comment]
Fixed in v3.3
